api.proto

```diff
@@ -214,21 +214,33 @@ message StreamMetadata {
         OK             = 0;
         UNKNOWN_STREAM = 1;
     }
-    string                        name       = 1; // The name of the stream being described
-    string                        subject    = 2; // The stream subject
-    Error                         error      = 3; // Indicates if there was something wrong with the requested stream
-    map<int32, PartitionMetadata> partitions = 4; // Information for the stream partitions
+    string                        name              = 1; // The name of the stream being described
+    string                        subject           = 2; // The stream subject
+    Error                         error             = 3; // Indicates if there was something wrong with the requested stream
+    map<int32, PartitionMetadata> partitions        = 4; // Information for the stream partitions
+    int64                         creationTimestamp = 5 [jstype=JS_STRING]; // When the stream has been created
+}
+
+// PartitionEventTimestamps contains timestamps related to a stream partition.
+message PartitionEventTimestamps {
+    int64 firstTimestamp  = 1 [jstype=JS_STRING]; // When the first event has occured
+    int64 latestTimestamp = 2 [jstype=JS_STRING]; // When the latest event has occured
 }
 
 // PartitionMetadata contains information for a stream partition.
 message PartitionMetadata {
-    int32            id            = 1; // Partition id
-    string           leader        = 2; // Broker id of the partition leader
-    repeated string  replicas      = 3; // Broker ids of the partition replicas
-    repeated string  isr           = 4; // Broker ids of the in-sync replica set
-    int64            highWatermark = 5; // High watermark of the partition
-    int64            newestOffset  = 6; // Newest offset of the partition
-    bool             paused        = 7; // Is this partition paused?
+    int32                    id                 = 1; // Partition id
+    string                   leader             = 2; // Broker id of the partition leader
+    repeated string          replicas           = 3; // Broker ids of the partition replicas
+    repeated string          isr                = 4; // Broker ids of the in-sync replica set
+    int64                    highWatermark      = 5; // High watermark of the partition
+    int64                    newestOffset       = 6; // Newest offset of the partition
+    bool                     paused             = 7; // Is this partition paused?
+    bool                     readonly           = 8; // Is this partition read-only?
+    int64                    creationTimestamp  = 9 [jstype=JS_STRING]; // When the partition has been created
+    PartitionEventTimestamps messageTimestamps  = 10; // Timestamps of when the first and latest message have been received
```

Comment:
Just a note on implementation for this: This will likely require the request be served by the leader of the partition since other brokers might not have this information since they aren't Raft-based. Is the plan to use `FetchPartitionMetadata` to populate this? See discussion [here](https://github.com/liftbridge-io/liftbridge/pull/242).

Reply:
Good point! Yes, this would be populated by `FetchPartitionMetadata`.